### PR TITLE
remove walt, support jwt

### PR DESCRIPTION
**Pull request by wistefan**
Handle verification with the trustbloc library.

### CODE REVIEW

File: verifier/jwt_verifier.go
Comment (pulledtim) [inline verifier/jwt_verifier.go:1]: Why did you add the impl from github.com/trustbloc/did-go ? was the impl buggy?

File: verifier/verifier.go [hunk 1]
```diff
@@ -252,7 +255,8 @@ func InitVerifier(config *configModel.Configuration, ssiKitClient ssikit.SSIKit)
 		common.JwtTokenSigner{},
 		credentialsConfig,
 		[]VerificationService{
-			&externalSsiKitVerifier,
+			// validation is now done with the trustbloc lib
+			//&externalSsiKitVerifier,
```
Comment (pulledtim) [inline verifier/verifier.go:259]: Not sure if I get that right, so the verifier will not call WaltId anymore right? I thought we are building on waltId for their current (and upcoming) support for did methods apart of web and key? The trustbloc lib will still need to cover the resolver-part to get the key/cert needed for validating the signature or?

Comment (wistefan) [reply verifier/verifier.go:259]: I removed it, since: 
- currently only web and key are relevant
- ssi-kit is retired and would require us to do a refactoring for using the new verison
- trustbloc is able to resolve web and key
- for new methods(most probably elsi and ebsi), we need to evaluate if its less work to update to the new walt-id solution or extend the trustbloc one

File: verifier/jwt_verifier.go
Comment (wistefan) [reply verifier/jwt_verifier.go:1]: Good question.  Less than two weeks ago and I already forgot... I will check

File: openapi/api_api.go [hunk 2]
```diff
@@ -172,7 +177,7 @@ func StartSIOPSameDevice(c *gin.Context) {
 
 // VerifierAPIAuthenticationResponse - Stores the credential for the given session
 func VerifierAPIAuthenticationResponse(c *gin.Context) {
-	state, stateExists := c.GetQuery("state")
+	state, stateExists := c.GetPostForm("state")
```
Comment (pulledtim) [inline openapi/api_api.go:180]: Wouldn't this be a breaking change?

File: verifier/gaiax.go [hunk 3]
```diff
@@ -44,17 +45,25 @@ func InitGaiaXRegistryVerificationService(verifierConfig *configModel.Verifier)
 	return verifier
 }
 
-func (v *GaiaXRegistryVerificationService) VerifyVC(verifiableCredential VerifiableCredential, verificationContext VerificationContext) (result bool, err error) {
-	if v.validateAll || slices.Contains(v.credentialTypesToValidate, verifiableCredential.GetCredentialType()) {
+func (v *GaiaXRegistryValidationService) ValidateVC(verifiableCredential *verifiable.Credential, verificationContext ValidationContext) (result bool, err error) {
```
Comment (pulledtim) [inline verifier/gaiax.go:48]: can we remove/rename verificationContext?

File: verifier/jwt_verifier.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+	return nil, ErrorNoVerificationKey
+}
+
+// the jwt-vc standard defines multiple options for the kid-header, while the standard implementation only allows for absolute paths.
+// see https://identity.foundation/jwt-vc-presentation-profile/#kid-jose-header
+// potential headers:
+//   - thePublicKey(1)
+//   - did:key:thePublicKey(2)
+//   - did:key:thePublicKey#id(3)
+func compareVerficationMethod(presentedMethod string, didDocumentMethod string) (result bool) {
```
Comment (pulledtim) [inline verifier/jwt_verifier.go:54]:
```suggestion
func compareVerificationMethod(presentedMethod string, didDocumentMethod string) (result bool) {
```

File: verifier/jwt_verifier.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+	if presentedMethod != "" {
+		return keyId == presentedMethod || absolutePath == presentedMethod || fullAbsolutePath == presentedMethod
+	}
+	logging.Log().Info("DidDocumentMethod is invalid.")
+	return false
+
+}
+
+func getKeyFromMethod(verficationMethod string) (keyId, absolutePath, fullAbsolutePath string, err error) {
```
Comment (pulledtim) [inline verifier/jwt_verifier.go:65]:
```suggestion
func getKeyFromMethod(verificationMethod string) (keyId, absolutePath, fullAbsolutePath string, err error) {
```

File: openapi/api_api.go (hunk 2 above)
Comment (wistefan) [reply openapi/api_api.go:180]: Yep. Thats why I used the chance to do it know, while we do a major upgrade anyways. Do you think its better to allow both for the time beeing?

File: verifier/trustedissuer.go [hunk 6]
```diff
@@ -3,94 +3,128 @@ package verifier
...
+			continue
+		}
+		if err != nil {
+			logging.Log().Warnf("Invalid til configured for type %s.", credentialType)
+			return false, err
+		}
+
+		tilAddress, credentialSupported := til[credentialType]
+		if !credentialSupported {
+			logging.Log().Debugf("No credential configured for type %s", credentialType)
```
Comment (pulledtim) [inline verifier/trustedissuer.go:62]: More like "no trusted issuer list configured for type" or?

File: openapi/api_api.go (hunk 2 above)
Comment (wistefan) [reply openapi/api_api.go]: done

File: verifier/gaiax.go (hunk 3 above)
Comment (wistefan) [reply verifier/gaiax.go]: done

File: verifier/jwt_verifier.go (hunk 5 above)
Comment (wistefan) [reply verifier/jwt_verifier.go]: done

**Review by pulledtim — APPROVED** [review]
